src/HealthChecks.Gcp.CloudFirestore/CloudFirestoreHealthCheck.cs

```diff
@@ -40,12 +42,14 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
                 return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
             }
         }
+
         private async Task<List<string>> GetRootCollectionsAsync(CancellationToken cancellationToken)
         {
             var collections = new List<string>();
 
             await foreach (var item in _cloudFirestoreOptions.FirestoreDatabase.ListRootCollectionsAsync())
             {
+                cancellationToken.ThrowIfCancellationRequested();

```

Comment:
I added this line to eliminate compiler warning about unused `cancellationToken` parameter. Alas, `ListRootCollectionsAsync` doesn't take `cancellationToken`.